### PR TITLE
Fix/cli error formatting

### PR DESCRIPTION
**Pull request by serialoverflow**
error.errors can be undefined which will lead to a TypeError, swallowing the actual error message in the process

Signed-off-by: Max Falk <gmdfalk@gmail.com>

## Hey, I just made a Pull Request!

Before, actual error message does not get displayed:
before

After, error message is shown:
after

You can reproduce/verify this issue by modifying the tsconfig.json in the backstage root directory with invalid syntax. `yarn build` will then always spit out a useless error message about error.errors. Spent too many hours tracking this down ...

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/cli/src/lib/builder/packager.ts [hunk 1]
```diff
@@ -22,21 +22,23 @@ import { paths } from '../paths';
...
   let msg = '';
 
   if (error.code === 'PLUGIN_ERROR') {
     if (error.plugin === 'esbuild') {
       msg += `${error.message}\n\n`;
-      for (const { text, location } of error.errors) {
-        const { line, column } = location;
-        const path = relativePath(paths.targetDir, error.id);
-        const loc = chalk.cyan(`${path}:${line}:${column}`);
+      if (error.errors) {
```
Comment (freben) [inline packages/cli/src/lib/builder/packager.ts:31]: Could we make this `if (error.errors?.length)` and also only add the double newlines if that condition is true? Just so we don't get the unnecessary whitespace.

**Review by freben — COMMENTED** [review]
Thanks, nice! Just one question

File: packages/cli/src/lib/builder/packager.ts (hunk 1 above)
Comment (serialoverflow) [reply packages/cli/src/lib/builder/packager.ts]: @freben Thanks! I've updated the code according to your suggestions.

**Review by freben — APPROVED** [review]

**Comment by freben** [comment]
I took the liberty to rebase this past a fix to the main branch that affected end-to-end tests through no fault of yours

**Comment by serialoverflow** [comment]
@freben: Great, thank you. I really appreciate how well-structured and easy the contribution process is.
